src/test/groovy/org/cyclonedx/gradle/utils/ExternalResourcesUtilTest.groovy

```diff
@@ -0,0 +1,50 @@
+package org.cyclonedx.gradle.utils
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.cyclonedx.gradle.TestUtils
+import org.cyclonedx.model.Bom
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import spock.lang.Specification
+
+class ExternalResourcesUtilTest extends Specification {
```

Comment:
The following test scenarios are required here:

1. Package without git reference
2. Package with VCS reference of different type, e.g. SVN
3. Package with incorrect git reference 
4. `git` executable not available on the system

Reply:
I added tests for most of what you've asked. Due to the fact that I am only intersted in the git VCS for now, I have ignored SVN altogether. In case a git url can't be resolved, the external reference will just not be added.